lib/extension.js

```diff
@@ -25,9 +25,9 @@ extension.log = function() {
  */
 extension.onMonitorEvent = function(event, path, stat) {
     /* jshint unused:false */
-    if (this.events.indexOf(event) > -1) {
+    if ( ! this._restarting && this.events.indexOf(event) > -1) {
+        this._restarting = true;
```

Comment:
I've seen the same guard in the luster itself, but it feels more natural to prevent a serial of restart attempts here.